### DISCUSSION TITLE
The idea of obtaining BTF directly from /sys/kernel/btf

### DISCUSSION THREAD

**Discussion by alobakin** [alobakin/linux#2] [Ideas]
Since 5.11, it's now possible to obtain a BTF not only for vmlinux, but also for modules.
This will eliminate a need for manually composing and registering a BTF inside the driver code, which is 100+ locs for ice for example.

That's obviously not the most straightforward and trivial way, but could help a lot.

**Comment by saeedtx** [comment]
the point of registering BTF directly from the driver is to allow "Flex metadata" meaning that meta data format can be constructed on the fly according to user demand.
BTF for modules is constructed only at compilation time and registered only on module load. so there is no way to implement flex metadata with vmlinux BTF. we still need a dynamic registration API for current and future HW where the HW will provide the BTF dynamically.

I am sure we can find mutliple ways to reduce the 100+ LOC, but the goal is to have the dynamic btf_register/unregister API
